CRM/Core/BAO/MessageTemplate.php

```diff
@@ -379,12 +381,16 @@ public static function sendTemplate($params) {
       }
     }
 
-    $subject = $dao->subject;
-    $text = $dao->text;
-    $html = $dao->html;
-    $format = $dao->format;
+    $mailContent = array(
+      'subject' => $dao->subject,
+      'text' => $dao->text,
+      'html' => $dao->html,
+      'format' => $dao->format,
+    );
     $dao->free();
 
+    CRM_Utils_Hook::alterMailContent($mailContent);
+
```

Comment:
Nope, w.r.t advantage of turning into `$subject`, definitely not. But I always feel variable imparts more clarity in readability than the array values. 

Anyway, this PR works fine in altering the mail params and content with these hooks. So, this changes can be merged as QA is being completed from my side.